src/nyc_trees/apps/home/training/types.py

```diff
@@ -210,38 +210,40 @@ def score(self, answers):
         return result
 
 
-class MultiChoiceQuestion(object):
+class Question(object):
     """
     Represents a question with multiple correct answers.
     text - string
     choices - iterable<string>
-    answer - iterable<int>; Indexes of the correct answers in choices iterable
+    answer - SingleAnswer or MultipleAnswer object
     """
     def __init__(self, text, choices, answer):
         assert len(choices) > 0
-        for ans in answer:
+        for ans in answer.answers:
             assert ans >= 0
             assert ans < len(choices)
         self.text = unicode(text)
         self.choices = list(choices)
-        self.answer = list(answer)
+        self.answer = answer
 
     def is_correct(self, candidate):
-        return candidate and set(candidate) == set(self.answer)
+        return candidate and set(candidate) == set(self.answer.answers)
 
-    def is_multiple_choice(self):
-        return True
+    def allow_multiple_selections(self):
```

Comment:
The rationale behind this name is that it's possible to allow multiple selections (checkbox) but only have 1 "answer". I think either name is an improvement though.